### PR TITLE
Cnnetantwo 60

### PR DESCRIPTION
**Pull request by dsover**
add breadcrumbs, fix routing, fix routing issues, update standards. This PR gets the application into a stable development state, and pulls together the work for the feature CNNETANTWO-57

Fixes: 
CNNETANTWO-57 (Feature) http://tickets.turner.com/browse/CNNETANTWO-57
CNNETANTWO-60 http://tickets.turner.com/browse/CNNETANTWO-60
CNNETANTWO-84 http://tickets.turner.com/browse/CNNETANTWO-84

screen shot 2017-07-26 at 9 59 53 am
screen shot 2017-07-26 at 10 00 15 am
screen shot 2017-07-26 at 10 00 40 am
screen shot 2017-07-26 at 10 00 49 am

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,6 +44,8 @@ Here are the common ones:
 
 - `npm test`: Runs unit tests
 
+- `npm run test-update-snapshots`: run the unit test and update jest snapshots, use this if you have intentionally changed the view of a component and need to update the snapshot for unit tests
```
Comment (jamsyoung) [inline README.md]: markdown lines should have a hard return at or before 80 columns.

This maintains a readable file when reading in the terminal.
(and is part of the markdown spec)

File: config.js [hunk 2]
```diff
@@ -18,7 +18,12 @@ let config = {
     brainApi: {
         uri: 'http://cnn-opp-brain-api.ref.services.ec2.dmtio.net'
     },
-    defaultAPITimeout_MS: 1000 * 5
+    defaultAPITimeout_MS: 1000 * 5,
+    routeMappings: { // these are the mappings for human readable breadcrumbs
+        '/': 'Home',
+        '/content-details': 'List',
+        '/list': 'List'
```
Comment (jamsyoung) [inline config.js]: Should there be two `List` items?

File: src/modules/breadcrumbs/index.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+class Breadcrumbs extends React.Component {
+
+    render() {
+        const history = this.props.history,
+            paths = ['/']; // initialize with a path to home
+
+        history.location.pathname.split('/').reduce((prev, curr, index) => {
+            paths[index] = (`${prev}/${curr}`)
+            return paths[index]
```
Comment (jamsyoung) [inline src/modules/breadcrumbs/index.js]: Does the .eslint allow no semi-colons?

Comment (dsover) [inline src/modules/breadcrumbs/index.js]: yeah it does, I can update it

Comment (jamsyoung) [inline src/modules/breadcrumbs/index.js]: it looks like there are several things I would have expected esilnt to catch. Noticed a few `if(...)` (no space after `if`)

File: README.md (hunk 1 above)
Comment (dsover) [inline README.md]: done

File: config.js (hunk 2 above)
Comment (dsover) [inline config.js]: Yeah it is fine to do this if we need to display the same name for different routes; however, they shouldn't be different routes. i updated so the route is now just /content

**Comment by jamsyoung** [comment]
The test directories are inconsistent.

```$ ls -R | perl -ne 'if(/(\/[^\/\-]*test[^\/]*\/)/){print $1."\n"}' | sort | uniq
/__test__/
/__tests__/
/test/
/tests/```

File: config.js [hunk 4]
```diff
@@ -18,7 +18,11 @@ let config = {
     brainApi: {
         uri: 'http://cnn-opp-brain-api.ref.services.ec2.dmtio.net'
     },
-    defaultAPITimeout_MS: 1000 * 5
+    defaultAPITimeout_MS: 1000 * 5,
```
Comment (khowen) [inline config.js]: Do we want this to be specific to API or should it be a static timeout interval? If specific to API, can we nest it inside above brainAPI{}?

File: src/modules/breadcrumbs/styles.css [hunk 5]
```diff
@@ -0,0 +1,21 @@
+
+ul.breadcrumb {
```
Comment (khowen) [inline src/modules/breadcrumbs/styles.css]: Do we want to use px?

File: src/modules/content-items/styles.css [hunk 6]
```diff
@@ -1,3 +1,3 @@
 .quarterWidth {
```
Comment (khowen) [inline src/modules/content-items/styles.css]: Do we want to rename this class? It's currently a "quarter," but we are making it a half?

File: src/modules/breadcrumbs/__test__/breadcrumbs.spec.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
```
Comment (jennyvallon) [inline src/modules/breadcrumbs/__test__/breadcrumbs.spec.js]: 'use strict'

Comment (jamsyoung) [inline src/modules/breadcrumbs/__test__/breadcrumbs.spec.js]: Are these tests passed through babel?  If so, then we don't use `use strict`.

File: src/modules/content-items/styles.css (hunk 6 above)
Comment (jamsyoung) [inline src/modules/content-items/styles.css]: YES, we should change the name if it doesn't reflect the value.

File: src/modules/breadcrumbs/__test__/breadcrumbs.spec.js (hunk 7 above)
Comment (dsover) [inline src/modules/breadcrumbs/__test__/breadcrumbs.spec.js]: Yes these tests go through bable-jest

File: src/modules/breadcrumbs/styles.css (hunk 5 above)
Comment (dsover) [inline src/modules/breadcrumbs/styles.css]: We are still deferring any style choices for design. this is just for POC. That said, CSS styles are not my strong suit, If you have any suggestions I will gladly take them.

**Pull request merged**